### PR TITLE
BH-1176: UCS - Webhooks

### PR DESCRIPTION
**Pull request by jmleroux**
### Before

srnt_webhook

### After

ucs_webhook

I had another solution in mind when starting the PR but it's  not possible for now to use the enveloppe in handler.
See this issue and conversation: https://github.com/symfony/symfony/issues/31075

### CODE REVIEW

**Review by willy-ahva — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleJsonHistory.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * @author JMLeroux <jean-marie.leroux@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ *
+ * A filesystem container to store Guzzle history in a json file.
+ * We need this filesytem json file to share it between process and keep track of the history in the tests
+ */
+class GuzzleJsonHistory implements \ArrayAccess, \Countable
+{
+    public function __construct(private string $filepath)
```
Comment (mmetayer) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleJsonHistory.php:20]: we could even use the local_storage flysystem adapter here

**Review by mmetayer — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleJsonHistory.php (hunk 1 above)
Comment (ahocquard) [reply src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleJsonHistory.php:20]: Or create a random file, which avoid to reset history

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleMockHandlerStack.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Response;
+
+/**
+ * @author JMLeroux <jean-marie.leroux@akeneo.com>
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ *
+ * A sub class of the Guzzle handler stack providing a JSON filesystem history.
+ * We need this handler to share its history between process and keep track of the history in the tests
```
Comment (ahocquard) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/GuzzleMockHandlerStack.php:18]:
```suggestion
 * The purpose of this handler is to share its history between a parent process and subprocess. It is useful to  keep track of the history of the webhook call in the tests. Indeed, the webhook call is performed in a subprocess.
```

**Review by ahocquard — APPROVED** [review]
Ok for now. I think it would be better to inject the context from the daemon later on as discussed 2 days ago.

**Pull request merged**